morphocluster/api.py

```diff
@@ -232,6 +235,81 @@ def get_file(path):
     )
 
 
+@api.route("/files/<path:path>", methods=["GET"])
+@api.route("/files/", methods=["GET"])
+def get_direntry(path=""):
+
+    directory_path = os.path.join(app.config["FILES_DIR"],path)
+
+    file_list = []
+
+    for entry in os.scandir(directory_path):
+        entry_info = {
+            "name": entry.name,
+            "path": os.path.relpath(entry.path, api.config["FILES_DIR"]),
+            "type": "directory" if entry.is_dir() else "file",
+            "last_modified": datetime.fromtimestamp(entry.stat().st_mtime).isoformat(),
+            "type_flag": 1 if entry.is_dir() else 0,
+        }
+        file_list.append(entry_info)
+
+    return jsonify(file_list)
+
+@api.route("/file/info/<path:file_path>", methods=["GET"])
+def get_file_info(file_path):
+    file_list = []
+    try:
+
+        file_path_on_server = os.path.join(app.config["FILES_DIR"], file_path)
+
+        if os.path.exists(file_path_on_server) and os.path.isfile(file_path_on_server):
+
+            file_info = {
+                "name": os.path.basename(file_path),
+                "path": file_path,
+                "type": "file",
+                "last_modified": datetime.fromtimestamp(os.path.getmtime(file_path_on_server)).isoformat(),
+            }
+            file_list.append(file_info)
+            return jsonify(file_list)
+        else:
+            return jsonify({"error": "File not found."}), 404
+    except Exception as e:
+        return jsonify({"error": str(e)}), 500
+    
+@api.route("/file/<path:file_path>", methods=["GET"])
+def get_file(file_path):
+    try:
+        file_path_on_server = os.path.join(app.config["FILES_DIR"], file_path)
+        if os.path.exists(file_path_on_server) and os.path.isfile(file_path_on_server):
+            return send_file(file_path_on_server, as_attachment=True)
+        else:
+            return jsonify({"error": "File not found."}), 404
+    except Exception as e:
+        return jsonify({"error": str(e)}), 500
+
+
+
+
+
+@api.route("/files/", methods=["POST"])
+@api.route("/files/<path:path>", methods=["POST"])
+def upload_file(path=""):
```

Comment:
`secure_file(path)`. No `try ... return 500`. `if not upload_file: raise werkzeug.exceptions.BadRequest()`.